lib/Locale/Maketext/Test.pm

```diff
@@ -0,0 +1,565 @@
+                         |
+                             ([\\'"?abfvntr])
+                         )
+                     /$1 ? chr(oct($1)) : $2 ? chr(hex($2)) : ($map{$3} || $3)/regx;
+}
+
+sub _bstring {
+    my @params;
+    return $_[0] =~ s!
+                         (?>                       # matches %func(%N,parameters...)
+                             %
+                             (?<func>\w+)
+                             \(
+                             %
+                             (?<p0>[0-9]+)
+                             (?<prest>[^\)]*)
+                             \)
+                         )
+                     |
+                         (?>                       # matches %func(parameters)
+                             %
+                             (?<simplefunc>\w+)
+                             \(
+                             (?<simpleparm>[^\)]*)
+                             \)
+                         )
+                     |                             # matches %N
+                         %
+                         (?<simple>[0-9]+)
+                     |                             # [, ] and ~ should be escaped as ~[, ~] and ~~
+                         (?<esc>[\[\]~])
+                     !
+                         if ($+{esc}) {
+                             "~$+{esc}";
+                         } elsif ($+{simplefunc}) {
+                             "[$+{simplefunc},$+{simpleparm}]";
+                         } else {
+                             my $pos = ($+{func} ? $+{p0} : $+{simple}) - 1;
+                             $params[$pos] = $+{func} && $+{func} eq 'plural' ? 'plural' : ($params[$pos] // 'text');
+                             $+{func} ? "[$+{func},_$+{p0}$+{prest}]" : "[_$+{simple}]";
+                         }
+                     !regx, \@params;
+}
+
+{
+    my @stack;
+
+    sub _nextline {
+        return pop @stack if @stack;
+        return scalar readline $_[0];
+    }
+
+    sub _unread {
+        push @stack, @_;
+    }
+}
+
+sub _get_trans {
+    my $f = shift;
+
+    while (defined(my $l = _nextline($f))) {
+        if ($l =~ /^\s*msgstr\s*"(.*)"/) {
+            my $line = $1;
+            while (defined($l = _nextline($f))) {
+                if ($l =~ /^\s*"(.*)"/) {
+                    $line .= $1;
+                } else {
+                    _unread($l);
+                    return _cstring($line);
+                }
+            }
+            return _cstring($line);
+        }
+    }
+}
+
+sub _get_po {
+    my ($self, $lang, $header_only) = @_;
+
+    unless ($lang =~ /\.po$/) {
+        $lang = $self->directory . '/' . $lang . '.po';
+    }
+
+    my (%header, @ids, $ln);
+    my $first = 1;
+
+    open my $f, '<:encoding(UTF-8)', $lang or die "Cannot open $lang: $!\n";
+    READ:
+    while (defined(my $l = _nextline($f))) {
+        if ($l =~ /^\s*msgid\s*"(.*)"/) {
+            my $line = $1;
+            $ln = $.;
```

Comment:
I think it is. But I think better to either add a comment why or to stash `$.` together with the line.

Here is why I think it is correct. A continuation line is marked with a leading space. Such lines are never un-read. When the program finds the next non-continuation, it un-reads it. At this point `$.` points at the un-read line. The next step is then to fetch the line again, this time from the stack. If it is a new beginning, `$.` is read. But it still has the accurate value. If it is not a new beginning, comment or empty line or junk, then the next line is read from the file which means `$.` is updated.

So, basically it boils down to yes it is accurate because the number of un-read lines is <=1.